### PR TITLE
Remove Some Assumptions from MultiRasterSource

### PR DESCRIPTION
**Pull request by jamesmcclain**
Optionally remove dtype and size assumptions.

In lieu of https://github.com/azavea/raster-vision/issues/1003

## Overview

Brief description of what this PR does, and why it is needed.

### Checklist

- [x] Updated `docs/changelog.rst`
- [x] Added `needs-backport` label if PR is bug fix that applies to previous minor release
- [x] Ran scripts/format_code and committed any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

Run the `analyze` and/or `chip` commands on a project that uses the MultiRasterSource and has sub-rasters of different dtypes and/or (slightly) different extents.

### CODE REVIEW

**Comment by AdeelH** [comment]
Why are we doing it this way instead of using transformers and enabling `use_transformers` everywhere?

File: rastervision_core/rastervision/core/data/raster_source/multi_raster_source.py
Comment (AdeelH) [inline rastervision_core/rastervision/core/data/raster_source/multi_raster_source.py:43]: Isn't the output of `RasterSource._get_chip()` supposed to be the same size as specified by the window it takes in?

Comment (jamesmcclain) [reply rastervision_core/rastervision/core/data/raster_source/multi_raster_source.py:43]: Well, the size of the overall chip ends up being the same size as the first subchip of the multiraster source.

In most instances, this will have no effect at all.  The idea here is to handle cases where there are multiple rasters which cover the same area but (in my case) differ by 1 in their width.  When that case is encountered, the extra column will be filled with the fill value in subchip from the smaller raster.

**Comment by jamesmcclain** [comment]
> Why are we doing it this way instead of using transformers and enabling `use_transformers` everywhere?

Good question.  There are two reasons:

The first is that I might eventually need raster transformers for all possible base types.  When running entirely locally those can be implemented in the pipeline, but if one wishes operate on Batch, custom images or upstream inclusion of those transformers is required.  This behavior seems to be simpler and more parsimonious than the full set of transformers.

The second reason is that I realized that I need behavior to handle slight differences in raster size due to reprojection, and that seemingly can only be handled at this level.  Since (I think) I need to make a change here, it made sense to make the other change that I need, as well.

File: rastervision_core/rastervision/core/data/raster_source/multi_raster_source.py
Comment (AdeelH) [reply rastervision_core/rastervision/core/data/raster_source/multi_raster_source.py:43]: Looking at the code for `RasterioSource`, it seems like `window` _is_ supposed to be in pixel coordinates and `RasterioSource` returns a chip of the height and width specified by the `window`.

Disabling the validation check for equal extents makes sense though.

**Comment by AdeelH** [comment]
> I might eventually need raster transformers for all possible base types

If by base types you mean dtypes, then a general `toDtype` transformer might be simpler. Alternatively, we could also make use of the existing `RasterSource.dtype` (or add a `target_dtype` property), and change `RasterSource` so that it casts to this `dtype` before returning chips. This will make each `RasterSource` responsible for its return type.

**Comment by jamesmcclain** [comment]
> > I might eventually need raster transformers for all possible base types
> 
> If by base types you mean dtypes, then a general `toDtype` transformer might be simpler. Alternatively, we could also make use of the existing `RasterSource.dtype` (or add a `target_dtype` property), and change `RasterSource` so that it casts to this `dtype` before returning chips. This will make each `RasterSource` responsible for its return type.

These sound like reasonable suggestions.

**Comment by lewfish** [comment]
These changes look ok to me.

**Comment by lewfish** [comment]
Adeel's approach may be more generic or elegant, but since this is already done, and there's a good chance MultiRasterSource will only be used on one project, I'm inclined to accept the change. We can revisit it later if it's too limited for what we need.

**Review by AdeelH — APPROVED** [review]

**Comment by jamesmcclain** [comment]
@lewfish Good to merge?  (I will try wait until after the CI issues are resolved.)

**Review by lewfish — APPROVED** [review]

**Pull request merged**
